server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapper.java

```diff
@@ -78,7 +78,14 @@ public RuleMapper(final Languages languages, final MacroInterpreter macroInterpr
   public Rules.Rule toWsRule(RuleDto ruleDto, SearchResult result, Set<String> fieldsToReturn) {
     Rules.Rule.Builder ruleResponse = Rules.Rule.newBuilder();
 
+    // Mandatory fields
     ruleResponse.setKey(ruleDto.getKey().toString());
+    Common.RuleType type = Common.RuleType.valueOf(ruleDto.getType());
+    if (type != null) {
```

Comment:
Why is the type in the list of mandatory fields ? It could be considered as optional and returned only if the parameter `f` has the value `type` (see other properties from line 89)